app/src/main/java/com/android/voyageur/model/trip/TripRepositoryFirebase.kt

```diff
@@ -2,66 +2,75 @@ package com.android.voyageur.model.trip
 
 class TripRepositoryFirebase(private val db: FirebaseFirestore) : TripRepository {
-  private val collectionPath = "trips"
+    private val collectionPath = "trips"
 
-  override fun getNewTripId(): String {
-    return db.collection(collectionPath).document().id
-  }
+    override fun getNewTripId(): String {
+        return db.collection(collectionPath).document().id
+    }
 
-  override fun init(onSuccess: () -> Unit) {
-    db.collection(collectionPath)
-        .get()
-        .addOnSuccessListener { onSuccess() }
-        .addOnFailureListener { exception ->
-          Log.e("TripRepositoryFirebase", "Error initializing repository: ", exception)
-        }
-  }
+    override fun init(onSuccess: () -> Unit) {
+        db.collection(collectionPath)
+            .get()
+            .addOnSuccessListener { onSuccess() }
+            .addOnFailureListener { exception ->
+                Log.e("TripRepositoryFirebase", "Error initializing repository: ", exception)
+            }
+    }
 
-  override fun getTrips(onSuccess: (List<Trip>) -> Unit, onFailure: (Exception) -> Unit) {
-    db.collection(collectionPath)
-        .get()
-        .addOnSuccessListener { result ->
-          val trips = result.map { document -> document.toObject(Trip::class.java) }
-          onSuccess(trips)
-        }
-        .addOnFailureListener { exception ->
-          Log.e("TripRepositoryFirebase", "Error getting trips: ", exception)
-          onFailure(exception)
-        }
-  }
+    override fun getTrips(onSuccess: (List<Trip>) -> Unit, onFailure: (Exception) -> Unit) {
+        db.collection(collectionPath)
+            .get()
+            .addOnSuccessListener { result ->
+                val trips = result.mapNotNull { document ->
+                    try {
+                        val trip = document.toObject(Trip::class.java)
+                        trip.copy(id = document.id) // Ensure the ID is set
+                    } catch (e: Exception) {
+                        Log.e("TripRepositoryFirebase", "Error parsing trip: ", e)
+                        null
+                    }
+                }
+                onSuccess(trips)
+            }
+            .addOnFailureListener { exception ->
+                Log.e("TripRepositoryFirebase", "Error getting trips: ", exception)
+                onFailure(exception)
+            }
+    }
 
-  override fun createTrip(trip: Trip, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
-    db.collection(collectionPath)
-        .document(trip.id)
-        .set(trip)
-        .addOnSuccessListener { onSuccess() }
-        .addOnFailureListener { exception ->
-          Log.e("TripRepositoryFirebase", "Error creating trip: ", exception)
-          onFailure(exception)
-        }
-  }
+    override fun createTrip(trip: Trip, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
+        db.collection(collectionPath)
+            .document(trip.id)
+            .set(trip, SetOptions.merge())
```

Comment:
Why is `SetOptions.merge()` useful here?

Reply:
The SetOptions.merge() method in Firebase Firestore is useful in this context to avoid overwriting existing fields in the document when you create or update a Trip.

Here’s why I used:

**1. Partial Updates:** When you use set() without any options, the document is fully replaced. This means that any fields in the existing document that are not present in the trip object will be removed. By using SetOptions.merge(), you ensure that only the fields from the trip object that are present get updated, and the other fields (if any) in the document remain untouched.
**2. Preserving Existing Data:** Suppose a document already exists with additional fields not included in the trip object you're passing. With merge(), the data in the document that is not in trip (like other properties or metadata) remains intact. This prevents accidental data loss.
**3. Flexible Document Structure:** When you allow dynamic additions to your document (for example, adding new fields over time), merge() ensures that these additions don't remove or replace other existing fields unless specified in the trip object. This is important when working with evolving or incomplete data models.

In short, SetOptions.merge() allows you to update or insert data into a document without overwriting the entire document, making it a safer option for partial updates in scenarios where the full structure of the document isn’t guaranteed or when you want to retain some existing fields.

Reply:
Not really useful, because the trip doesn't exist by desgin

Reply:
Using SetOptions.merge() here isn't strictly necessary since this operation is for creating a new trip. When creating a document, there are no existing fields to preserve, so a regular set() would work just as well. However, I'll go ahead and fix it to remove the merge option, as it's more appropriate in this case. Thanks for pointing that out!